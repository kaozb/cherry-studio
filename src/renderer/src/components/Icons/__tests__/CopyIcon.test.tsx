import { render } from '@testing-library/react'
import { describe, expect, it, vi } from 'vitest'

import CopyIcon from '../CopyIcon'

describe('CopyIcon', () => {
  it('should match snapshot with props and className', () => {
    const onClick = vi.fn()
    const { container } = render(
      <CopyIcon className="custom-class" onClick={onClick} title="Copy to clipboard" data-testid="copy-icon" />
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
