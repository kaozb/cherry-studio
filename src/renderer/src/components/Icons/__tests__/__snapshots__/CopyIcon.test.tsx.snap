// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CopyIcon > should match snapshot with props and className 1`] = `
<i
  class="iconfont icon-copy custom-class"
  data-testid="copy-icon"
  title="Copy to clipboard"
/>
`;
